src/goto-checker/all_properties_verifier.h

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************\
+
+Module: Goto Verifier for Verifying all Properties
+
+Author: Daniel Kroening, Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// Goto Verifier for Verifying all Properties
+
+#ifndef CPROVER_GOTO_CHECKER_ALL_PROPERTIES_VERIFIER_H
+#define CPROVER_GOTO_CHECKER_ALL_PROPERTIES_VERIFIER_H
+
+#include "goto_verifier.h"
+
+#include "incremental_goto_checker.h"
+#include "properties.h"
+#include "report_util.h"
+
+template <class incremental_goto_checkerT>
+class all_properties_verifiert : public goto_verifiert
+{
+public:
+  all_properties_verifiert(
+    const optionst &options,
+    ui_message_handlert &ui_message_handler,
+    abstract_goto_modelt &goto_model)
+    : goto_verifiert(options, ui_message_handler),
+      goto_model(goto_model),
+      incremental_goto_checker(options, ui_message_handler, goto_model)
+  {
+    properties = std::move(initialize_properties(goto_model));
```

Comment:
I think you don't need `std::move` with an unnamed temporary, but I could be wrong.

Reply:
Yes, this should work without move; there is only a single allocation site in the callee.